### PR TITLE
Fix showcase components TS errors 

### PR DESCRIPTION
**Pull request by frederickobrien**
This PR fixes TS issues for components on [the support showcase](https://support.theguardian.com/eu/support), plus a few others impacted as a knock-on effect. Formatting best practice and explicit return types have also been added to edited files.

### CODE REVIEW

File: support-frontend/assets/pages/showcase/components/hero.tsx [hunk 1]
```diff
@@ -41,15 +47,17 @@ const getCountrySelector = (countryGroupId: CountryGroupId) => {
...
 	const countrySelector = getCountrySelector(props.countryGroupId);
 	return (
 		<div className="showcase-hero">
 			<figure className="showcase-hero-wrapper">
 				<div className="showcase-hero-heading">
-					<h1 className="visually-hidden">Support the guardian</h1>
+					<h1 className="visually-hidden">Support the Guardian</h1>
 					<h2 className="visually-hidden">
-						Help us support investigative independent journalism
+						Help us deliver investigative independent journalism
```
Comment (frederickobrien) [inline support-frontend/assets/pages/showcase/components/hero.tsx:60]: I've changed this to match what's displayed on [the support showcase page](https://support.theguardian.com/eu/support):

image

File: support-frontend/assets/pages/showcase/components/hero.tsx [hunk 2]
```diff
@@ -41,15 +47,17 @@ const getCountrySelector = (countryGroupId: CountryGroupId) => {
...
 	const countrySelector = getCountrySelector(props.countryGroupId);
 	return (
 		<div className="showcase-hero">
 			<figure className="showcase-hero-wrapper">
 				<div className="showcase-hero-heading">
-					<h1 className="visually-hidden">Support the guardian</h1>
+					<h1 className="visually-hidden">Support the Guardian</h1>
 					<h2 className="visually-hidden">
-						Help us support investigative independent journalism
+						Help us deliver independent investigative journalism
```
Comment (frederickobrien) [inline support-frontend/assets/pages/showcase/components/hero.tsx:60]: I've changed this to match what's displayed on [the support showcase page](https://support.theguardian.com/eu/support):

image

File: support-frontend/assets/pages/showcase/components/hero.tsx [hunk 3]
```diff
@@ -13,13 +14,18 @@ import HeroImg from './hero.svg';
...
-const Caption = (props: { className: string; captionText: string }) => (
-	<div className={props.className}>
-		<figcaption className="showcase-hero__caption">
-			<p>{props.captionText}</p>
-		</figcaption>
-	</div>
-);
+function Caption(props: {
+	className: string;
+	captionText: EmotionJSX.Element;
```
Comment (frederickobrien) [inline support-frontend/assets/pages/showcase/components/hero.tsx:19]: Captions are stored in [`regionalContent.tsx`](https://github.com/guardian/support-frontend/blob/ee1e90f7f83fb911f02238445ae35da568e67dce/support-frontend/assets/pages/showcase/components/regionalContent.tsx) as JSX Elements, not as strings. The current support showcase has a <span> element inside the <p> rendered here, so this change seems consistent with that.

image

I suppose the alternative would be to rejig captions in `regionalContent.tsx` to be actual strings?

File: support-frontend/assets/components/button/anchorButton.tsx [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
 	href: string;
+	onClick?: () => void;
 };
 
-const AnchorButton = (props: PropTypes) => (
-	<SharedButton element="a" {...props} />
-);
+function AnchorButton(props: PropTypes): EmotionJSX.Element {
+	return <SharedButton element="a" {...props} />;
+}
```
Comment (TomPretty) [inline support-frontend/assets/components/button/anchorButton.tsx:16]: I think we've been standardising on `JSX.Element` as the return value from function components. It's compatible with the `EmotionJSX.Element` type so it should be straightforward to swap it out with `JSX.Element`

File: support-frontend/assets/pages/showcase/components/hero.tsx (hunk 3 above)
Comment (TomPretty) [inline support-frontend/assets/pages/showcase/components/hero.tsx:19]: Yeah I think it's fine to have this as JSX. I'd say change the type to `JSX.Element` though

File: support-frontend/assets/components/button/anchorButton.tsx (hunk 4 above)
Comment (frederickobrien) [reply support-frontend/assets/components/button/anchorButton.tsx]: Ah yep, I was blindly following what VSC suggested. Thanks Tom. Done!

**Review by TomPretty — APPROVED** [review]
Nice! 🚀

**Pull request merged**